### DISCUSSION TITLE
Adding the AirBnb ESLint rules ?

### DISCUSSION THREAD

**Discussion by tofl** [nuxt/nuxt#7836] [General]
Hi,

I'm starting out with NuxtJS, coming directly from VueJS where I use (and love) the AirBnb ESLint coding rules. I tried installing it in a NuxtJS project but noticed it's a bit more complicated than I thought. Since those rules are not available to be pre-installed when creating a Nuxt project, I'm starting to think that the AirBnb rules go against Nuxt's coding rules. Am I mistaken?

Thanks.

**Comment by TheAlexLichter** [comment]
Hey 👋 

The "official" [eslint-config](https://github.com/nuxt/eslint-config) won't support AirBnb rules right now (see https://github.com/nuxt/eslint-config/issues/41) but feel free to use your own preset as you like :)

We don't force you to use the "preset" ESLint rules. So it shouldn't be a problem to add the preset instead of the nuxt config one :)

**Reply by tofl** [reply]
Okay then. And if I may make a suggestion, it would be nice to offer the same presets when using `npx create-nuxt-app` than when doing `vue create`.

**Reply by TheAlexLichter** [reply]
> 
> 
> Okay then. And if I may make a suggestion, it would be nice to offer the same presets when using `npx create-nuxt-app` than when doing `vue create`.

Feel free to suggest that in the [create-nuxt-app repo](https://github.com/nuxt/create-nuxt-app) ☺️

**Comment by markabruce** [comment]
I understand this was some months ago, but for anyone wanting a fairly quick win in using AirBnb ESLint rules instead, I have a Medium post [here](https://medium.com/@markabrucey/configuring-nuxtjs-to-use-airbnb-eslint-with-vuetify-d56eedf14f3a?source=friends_link&sk=a36c6f5c778543ff1a547cab60a22861).

I hope that helps.

**Reply by tofl** [reply]
That's cool, I'll have a look thanks

**Comment by JasonCubic** [comment]
A quick and easy way to add airbnb rules.  I haven't run into any problems doing it this way yet.

add the repo that has the rules.  `npm i -D eslint-config-airbnb-base`

then just add the airbnb rules to your .eslintrc.js

```
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true,
  },
  parserOptions: {
    parser: 'babel-eslint',
  },
  extends: [
    'plugin:vue/essential',
    './node_modules/eslint-config-airbnb-base/rules/best-practices.js',
    './node_modules/eslint-config-airbnb-base/rules/errors.js',
    './node_modules/eslint-config-airbnb-base/rules/es6.js',
    './node_modules/eslint-config-airbnb-base/rules/imports.js',
    './node_modules/eslint-config-airbnb-base/rules/node.js',
    // './node_modules/eslint-config-airbnb-base/rules/strict.js',
    './node_modules/eslint-config-airbnb-base/rules/style.js',
    './node_modules/eslint-config-airbnb-base/rules/variables.js',
  ],
  plugins: [
    'vue',
  ],
  // add your custom rules here
  rules: {},
};

```

**Reply by samydoesit** [reply]
Nice solution.
Did you also manage to crash the build on lint errors?

**Reply by JasonCubic** [reply]
Seems to build okay.  Maybe I don't understand exactly what you are asking?

**Reply by samydoesit** [reply]
Got it, adding these did the trick:
```javascript
extends: [
    ...,
     'plugin:import/errors',
    'plugin:import/warnings',
  ]
```

If you have some lint error your nuxt build on ` npm run dev` or build will crash.

**Reply by JasonCubic** [reply]
ah, yes, the imports.  I should have mentioned that.  I installed eslint-import-resolver-alias and then configured the settings in the .eslintrc.js file to look like this:
```
  settings: {
    'import/resolver': {
      alias: {
        map: [
          ['~', __dirname],
          ['@', __dirname],
        ],
        extensions: ['.vue'],
      },
    },
  },
```

if your method works better though, it seems simpler.
